packages/blockchain-extension/extension/commands/exportConnectionProfileCommand.ts

```diff
@@ -51,7 +51,8 @@ export async function exportConnectionProfile(gatewayTreeItem: GatewayTreeItem,
     // Ask the user where they want to export it to
     // set the default path to be the first open workspace folder
     let defaultPath: string;
-    const fileName: string = `${gatewayEntry.name}_connection.json`;
+    let fileName: string = `${gatewayEntry.name}_connection.json`;
+    fileName = fileName.replace(/\s/g, '');
```

Comment:
This removed white spaces, but does not create a upper first camel case as requested.

I created a gateway with the following terrible name: _attempt 1 my one-org local fabric GATEWAY_

The current code will create a file name that is very hard to read: `attempt 1 my one-org local fabric GATEWAY_connection.json`

In upper camel case (where all words start with upper case, followed by lower case, hyphens and spaces are ignored, and numbers are not touched), this would be much more readable: `Attempt1MyOneOrgLocalFabricGatewayConnection.json`

You can achieve this with _upperFirst_ and _camelCase_ from the lodash package for example.